source/beamline/SXRMB/SXRMBBeamline.cpp

```diff
@@ -670,32 +579,26 @@ void SXRMBBeamline::setupComponents()
-	VVR16065B1001Valve_ = new CLSExclusiveStatesControl("VVR16065B1001", "VVR1606-5-B10-01:state", "VVR1606-5-B10-01:opr:open", "VVR1606-5-B10-01:opr:close", this, "VVR1606-5-B10-01 Valve");
+	beamlineShutters_->setSafetyShutter(SSH1406B1001Shutter, CLSExclusiveStatesControl::Open, CLSExclusiveStatesControl::Closed);
+	beamlineShutters_->addShutter(PSH1406B1002Shutter_, CLSExclusiveStatesControl::Open, CLSExclusiveStatesControl::Closed, 1);
 
-	beamlineControlShutterSet_ = new AMControlSet(this);
-	beamlineControlShutterSet_->addControl(SSH1406B1001Shutter_);
-	beamlineControlShutterSet_->addControl(PSH1406B1002Shutter_);
-	beamlineControlShutterSet_->addControl(VVR16064B1003Valve_);
-	beamlineControlShutterSet_->addControl(VVR16064B1004Valve_);
-	beamlineControlShutterSet_->addControl(VVR16064B1006Valve_);
-	beamlineControlShutterSet_->addControl(VVR16064B1007Valve_);
-	beamlineControlShutterSet_->addControl(VVR16065B1001Valve_);
+	// the valves used for Beam on/off control
+	CLSExclusiveStatesControl* VVR16064B1003Valve = new CLSExclusiveStatesControl("VVR16064B1003", "VVR1606-4-B10-03:state", "VVR1606-4-B10-03:opr:open", "VVR1606-4-B10-03:opr:close", this, "VVR1606-4-B10-03 Valve");
+	CLSExclusiveStatesControl* VVR16064B1004Valve = new CLSExclusiveStatesControl("VVR16064B1004", "VVR1606-4-B10-04:state", "VVR1606-4-B10-04:opr:open", "VVR1606-4-B10-04:opr:close", this, "VVR1606-4-B10-04 Valve");
+	// NOT THIS ONE! It's connected to the pump on the mono
+	//CLSBiStateControl *VVR16064B1005Valve;
+	CLSExclusiveStatesControl* VVR16064B1006Valve = new CLSExclusiveStatesControl("VVR16064B1006", "VVR1606-4-B10-06:state", "VVR1606-4-B10-06:opr:open", "VVR1606-4-B10-06:opr:close", this, "VVR1606-4-B10-06 Valve");
+	CLSExclusiveStatesControl* VVR16064B1007Valve = new CLSExclusiveStatesControl("VVR16064B1007", "VVR1606-4-B10-07:state", "VVR1606-4-B10-07:opr:open", "VVR1606-4-B10-07:opr:close", this, "VVR1606-4-B10-07 Valve");
+	CLSExclusiveStatesControl* VVR16065B1001Valve = new CLSExclusiveStatesControl("VVR16065B1001", "VVR1606-5-B10-01:state", "VVR1606-5-B10-01:opr:open", "VVR1606-5-B10-01:opr:close", this, "VVR1606-5-B10-01 Valve");
 
-	beamlineShutters_->addShutter(SSH1406B1001Shutter_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-	beamlineShutters_->addShutter(PSH1406B1002Shutter_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-
-	beamlineValves_->addValve(VVR16064B1003Valve_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-	beamlineValves_->addValve(VVR16064B1004Valve_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-	beamlineValves_->addValve(VVR16064B1006Valve_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-	beamlineValves_->addValve(VVR16064B1007Valve_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
-	beamlineValves_->addValve(VVR16065B1001Valve_, CLSBEAMLINE_VALVE_OPEN, CLSBEAMLINE_VALVE_CLOSED);
+	beamlineValves_->addValve(VVR16064B1003Valve, CLSExclusiveStatesControl::Open, CLSExclusiveStatesControl::Closed, 1);
```

Comment:
@dretrex there are waits actions. This is not a frequently used feature. Usually, the valves are all open. To put valves in the beam on actions is to make sure the beam will be ON if they want.